app/models/katello/repository.rb

```diff
@@ -20,10 +20,14 @@ class Repository < Katello::Model
     FILE_TYPE = 'file'
     PUPPET_TYPE = 'puppet'
     DOCKER_TYPE = 'docker'
-    TYPES = [YUM_TYPE, FILE_TYPE, PUPPET_TYPE, DOCKER_TYPE]
-    SELECTABLE_TYPES = [YUM_TYPE, PUPPET_TYPE, DOCKER_TYPE]
     CHECKSUM_TYPES = %w(sha1 sha256)
 
+    def self.creatable_repository_types
+      Katello::ContentPlugin_Manager.repository_types.select do |repo_type|
+        Katello::ContentPlugin_Manager.creatable_by_user?(repo_type)
+      end
+    end
```

Comment:
Also, it looks like the class is ContentPluginManager  not ContentPlugin_Manager